src/consumer.ts

```diff
@@ -527,3 +527,9 @@ export class Consumer extends TypedEventEmitter {
     }
   }
 }
+
+function wrapUnexpectedErrorMessage(err: Error) {
```

Comment:
A delayed review, but I'm still having a think on this function and if this is something we want to do or not.

It adds a lot of complexity and is a bit of a hack. If Zod requires this sort of hack, then I'm not sure that we want to support Zod implementations.

Reply:
Is there a workaround I can use if I really need to use a Zod implementation but I also need to see the error messages in production?

Reply:
👋 Quick suggestion: since the library already requires [Node 16+](https://node.green/#ES2022-features-Error-cause-property), you can switch the error handling to something like:

```
if (err instanceof Error) {
  throw new SQSError('Unexpected message handler failure', { cause: err })
}
```

IIRC you have an `SQSError` type, which if used here would allow users of this library to easily detect that the error came from the library and unwrap the original error if necessary.

With that, Node will automatically combine both error messages and chain the stack traces when serializing the error. As far as best practices go, this would be the best you can get. 

As for mutating the original error object thrown by the "user", that is already quite "hacky" and I would highly recommend moving away from that approach. 

`cause` is available since ES2022 and according to [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) is compatible with all major browsers.

The caveat is that some tools (like IntelliJ if you're running jest tests) may not still properly support showing the `cause` attribute. Also, if there are many items added to the final stack trace, Node may truncate it too early, since IIRC it limits stack trace printing up to 10 items by default, which can be adjusted with something like `node --stack-trace-limit=15`.

Reply:
Yeah this part of the code is probably quite old, it pre dates my involvement any way.

We can look into using cause, my main problem with the solution in this PR is the catch that goes no where.

If cause solves the issue with it not working with zod and still allows for identification, I'd be fine with that change, we'd just have to note it as potentially breaking.

Reply:
I've done some changes in https://github.com/bbc/sqs-consumer/pull/414/commits/81a7b5c9c7e73a3de7cbfb7f3e7e8051d2313d44 to use new `toError` functions, the tests are still passing, so either this works with the presumed fix or the tests are wrong, tests probably need looking at.